modin/backends/base/query_compiler.py

```diff
@@ -33,11 +32,10 @@
 
 
 def _get_axis(axis):
-    def axis_getter(self):
```

Comment:
I don't think we should change the implementation of this method because the query compiler located here is responsible for default to pandas execution. I suggest keeping `_get_axis` methods here and in pandas QC as those were.

Reply:
In that case we should fix https://github.com/modin-project/modin/blob/e633a32c2706f75fbbdf852deace9e3384b31086/modin/engines/base/frame/data.py#L2219-L2242
because, I think, root cause of issue is placed here. In `PandasFrame.to_pandas` method in case of empty df we update only `columns`, but not `index`, because for empty `Series` `PandasFrame.columns` is equal to [`__reduced__`]. In this case, when we created empty `Series` with specific type of `index` (datetime), we have correct type for internal `PandasFrame.index`, but we will have default type of `index` for `pandas.DataFrame`, that will returned from `PandasFrame.to_pandas`. Possible, this is actual for the other backends.